vms/avm/vm.go

```diff
@@ -483,26 +483,85 @@ func (vm *VM) GetUTXOs(
 		limit = maxUTXOsToFetch
 	}
 
-	seen := ids.Set{} // IDs of UTXOs already in the list
+	if paginate {
+		return vm.getPaginatedUTXOs(addrs, startAddr, startUTXOID, limit)
+	}
+	return vm.getAllUTXOs(addrs, startAddr, startUTXOID)
+}
+
+func (vm *VM) getPaginatedUTXOs(addrs ids.ShortSet,
+	startAddr ids.ShortID,
+	startUTXOID ids.ID,
+	limit int,
+) ([]*avax.UTXO, ids.ShortID, ids.ID, error) {
+	seen := ids.New(limit) // IDs of UTXOs already in the list
 	utxos := make([]*avax.UTXO, 0, limit)
 	lastAddr := ids.ShortEmpty
 	lastIndex := ids.Empty
 	addrsList := addrs.List()
 	ids.SortShortIDs(addrsList)
+
+	for _, addr := range addrsList {
```

Comment:
I'm not sure that this loop is going to work as expected. If we were to set this up such that:

- We had two addresses: `A`, and `B`.
- We had two UTXOs: `X`, and `Y` where `X` referenced both `A` and `B` and `Y` referenced `B`.
- We call this function with a limit 2.

Won't this function return just `[X]`? Rather than the expected `[X, Y]` (Assuming the addresses and UTXOs are ordered lexicographically)

Reply:
I don't think so 🤔  

```
Addresses [A{X, Y}, B{Y}]
Iteration 1
- Address A
- X appends
- Y appends
- Limit hits 
- break and return
```

```
Addresses [A{X, Y}, B{Y}]
Iteration 1
- Address B
- Y appends
Iteration 2
- Address A
- X appends
- Limit hits 
- break and return
```

I'm not sure going B->A is doable but it looks ok - Am I missing it ? :)

Reply:
Sorry yeah I was trying to explain the limit bug, but I over simplified my example. You seem to have caught it regardless.